### ISSUE TITLE
tests: Validate Clang 11

### ISSUE THREAD

**Issue by StephanTLavavej** [microsoft/STL#1027]
The Clang/LLVM devs [are planning](https://lists.llvm.org/pipermail/cfe-dev/2020-July/066191.html) to branch for the release of Clang 11 on July 15, 2020.

It would be nice to validate the STL's tests against Clang 11, so if any new warnings or errors have appeared, we can either clean them up in the STL ahead of time, or report bugs to the Clang devs so they can be fixed before release (Clang is really awesome about fixing bugs quickly, but we have to report them in time).

**Comment by StephanTLavavej** [comment]
Clang 11 RC1 has been tagged and pre-built binaries will be available soon: https://prereleases.llvm.org/11.0.0/

**Comment by CaseyCarter** [comment]
I plugged LLVM 11 into the internal test harness and ran the stl tests flawlessly. Assuming that I'd made a mistake and that the new Clang wasn't being picked up, I made Clang 11 required in `yvals_core.h` and ran the stl tests flawlessly again.

TLDR: It looks like we are fine with Clang 11 and can expect no regressions on release. (Certainly nothing like the mess when Clang 10 enabled concepts and the entire library caught on fire.)

**Comment by CaseyCarter** [comment]
> I plugged LLVM 11.0.0-rc1 into the internal test harness and ran the stl tests flawlessly. Assuming that I'd made a mistake and that the new Clang wasn't being picked up, I made Clang 11 required in `yvals_core.h` and ran the stl tests flawlessly again.

Ditto LLVM 11.0.0-rc2.

**Comment by CaseyCarter** [comment]
> > I plugged LLVM 11.0.0-rc1 into the internal test harness and ran the stl tests flawlessly. Assuming that I'd made a mistake and that the new Clang wasn't being picked up, I made Clang 11 required in `yvals_core.h` and ran the stl tests flawlessly again.
> 
> Ditto LLVM 11.0.0-rc2.

Ditto LLVM 11.0.0-rc3. (Which must have been somehow tested by cobbler elves because I certainly couldn't have done it while I'm on vacation.)

**Comment by CaseyCarter** [comment]
> Ditto LLVM 11.0.0-rc3. (Which must have been somehow tested by cobbler elves because I certainly couldn't have done it while I'm on vacation.)

Ditto LLVM 11.0.0-rc4.
